File: ApiExamples/CSharp/ExSavingCallback.cs
using System.IO;
using Aspose.Words;
using Aspose.Words.Saving;
using NUnit.Framework;

namespace ApiExamples
{
    [TestFixture]
    internal class ExSavingCallback : ApiExampleBase
    {
        [Test]
        public void CheckThatAllMethodsArePresent()
        {
            HtmlFixedSaveOptions htmlFixedSaveOptions = new HtmlFixedSaveOptions();
            htmlFixedSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            ImageSaveOptions imageSaveOptions = new ImageSaveOptions(SaveFormat.Png);
            imageSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            PdfSaveOptions pdfSaveOptions = new PdfSaveOptions();
            pdfSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            PsSaveOptions psSaveOptions = new PsSaveOptions();
            psSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            SvgSaveOptions svgSaveOptions = new SvgSaveOptions();
            svgSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            SwfSaveOptions swfSaveOptions = new SwfSaveOptions();
            swfSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            XamlFixedSaveOptions xamlFixedSaveOptions = new XamlFixedSaveOptions();
            xamlFixedSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            XpsSaveOptions xpsSaveOptions = new XpsSaveOptions();
            xpsSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();
        }

        [Test]
        public void PageFileNameSavingCallback()
        {
            //ExStart
            //ExFor:IPageSavingCallback
            //ExFor:HtmlFixedSaveOptions.PageSavingCallback
            //ExSummary:Shows how separate pages are saved when a document is exported to fixed page format.
            Document doc = new Document(MyDir + "Rendering.doc");

            HtmlFixedSaveOptions htmlFixedSaveOptions = new HtmlFixedSaveOptions { PageIndex = 0, PageCount = doc.PageCount };
            htmlFixedSaveOptions.PageSavingCallback = new CustomPageFileNamePageSavingCallback();

            doc.Save(MyDir + @"\Artifacts\Rendering.html", htmlFixedSaveOptions);

            string[] filePaths = Directory.GetFiles(MyDir + @"\Artifacts\", "Page_*.html");

            for (int i = 0; i < doc.PageCount; i++)
            {
                string file = string.Format(MyDir + @"\Artifacts\Page_{0}.html", i);
                Assert.AreEqual(file, filePaths[i]);//ExSkip
            }
        }

        /// <summary>
        /// Custom PageFileName is specified.
        /// </summary>
        private class CustomPageFileNamePageSavingCallback : IPageSavingCallback
        {
            public void PageSaving(PageSavingArgs args)
            {
                // Specify name of the output file for the current page.
                args.PageFileName = string.Format(MyDir + @"\Artifacts\Page_{0}.html", args.PageIndex);
            }
        }
        //ExEnd
    }
}

File: ApiExamples/VisualBasic/ExSavingCallback.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Words
Imports Aspose.Words.Saving
Imports NUnit.Framework

Namespace ApiExamples
	<TestFixture> _
	Friend Class ExSavingCallback
		Inherits ApiExampleBase
		<Test> _
		Public Sub CheckThatAllMethodsArePresent()
			Dim htmlFixedSaveOptions As New HtmlFixedSaveOptions()
			htmlFixedSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			Dim imageSaveOptions As New ImageSaveOptions(SaveFormat.Png)
			imageSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			Dim pdfSaveOptions As New PdfSaveOptions()
			pdfSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			Dim psSaveOptions As New PsSaveOptions()
			psSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			Dim svgSaveOptions As New SvgSaveOptions()
			svgSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			Dim swfSaveOptions As New SwfSaveOptions()
			swfSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			Dim xamlFixedSaveOptions As New XamlFixedSaveOptions()
			xamlFixedSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			Dim xpsSaveOptions As New XpsSaveOptions()
			xpsSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()
		End Sub

		<Test> _
		Public Sub PageFileNameSavingCallback()
			Dim doc As New Document(MyDir & "Rendering.doc")

			Dim htmlFixedSaveOptions As HtmlFixedSaveOptions = New HtmlFixedSaveOptions With {.PageIndex = 0, .PageCount = doc.PageCount}
			htmlFixedSaveOptions.PageSavingCallback = New CustomPageFileNamePageSavingCallback()

			doc.Save(MyDir & "\Artifacts\out.html", htmlFixedSaveOptions)

			Dim filePaths() As String = Directory.GetFiles(MyDir, "Page_*.html")

			For i As Integer = 0 To doc.PageCount - 1
				Dim file As String = String.Format(MyDir & "Page_{0}.html", i)
				Assert.AreEqual(file, filePaths(i))
			Next i
		End Sub

		<Test> _
		Public Sub PageStreamSavingCallback()
			Dim docStream As Stream = New FileStream(MyDir & "Rendering.doc", FileMode.Open)
			Dim doc As New Document(docStream)

			Dim htmlFixedSaveOptions As HtmlFixedSaveOptions = New HtmlFixedSaveOptions With {.PageIndex = 0, .PageCount = doc.PageCount}
			htmlFixedSaveOptions.PageSavingCallback = New CustomPageStreamPageSavingCallback()

			doc.Save(MyDir & "\Artifacts\out.html", htmlFixedSaveOptions)

			docStream.Close()
		End Sub

		''' <summary>
		''' Custom PageFileName is specified.
		''' </summary>
		Private Class CustomPageFileNamePageSavingCallback
			Implements IPageSavingCallback
            Public Sub PageSaving(ByVal args As PageSavingArgs) Implements IPageSavingCallback.PageSaving
                ' Specify name of the output file for the current page.
                args.PageFileName = String.Format(MyDir & "Page_{0}.html", args.PageIndex)
            End Sub
		End Class

		''' <summary>
		''' Custom PageStream is specified.
		''' </summary>
		Private Class CustomPageStreamPageSavingCallback
			Implements IPageSavingCallback
            Public Sub PageSaving(ByVal args As PageSavingArgs) Implements IPageSavingCallback.PageSaving
                ' Specify memory stream for the current page.
                args.PageStream = New MemoryStream()
                args.KeepPageStreamOpen = True
            End Sub
		End Class
	End Class
End Namespace
